api/listener.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// ListenAndServe starts the REST API
+func (a *API) ListenAndServe(ctx context.Context, hostAndPort string) {
+	baseCtx, cancel := context.WithCancel(context.Background())
+
+	log := logrus.WithField("component", "api")
+
+	server := &http.Server{
+		Addr:              hostAndPort,
+		Handler:           a.handler,
+		ReadHeaderTimeout: 2 * time.Second, // to mitigate a Slowloris attack
```

Comment:
A little fuzzy here so going to restate my current understanding to clarify:

This blocks a Slow Lorris attack by putting a limit on the total amount of time taken to read headers. 
In a slow Lorris attack, multiple connections are created and partial headers are repeatedly sent to prolong each connection. 

However, this doesn't work with the `ReadHeaderTimeout` as connection will timeout after 2 seconds and force a release of the connection regardless  of how many partial headers are sent.

Reply:
Yes exactly. BTW this is an issue that `gosec` finds all the time, although it's not applicable with GoTrue. The reason why it is not applicable is that typically GoTrue is behind some sort of proxy -- be it NGINX, load balancer, CDN... all of those handle the "slow sender" issues... i.e. they wait for a sizable chunk of the body to be received before sending the whole package at once to the origin (GoTrue).